From 22ca4c068c52aa3fa155d7bb7d60a1e7cd127940 Mon Sep 17 00:00:00 2001
From: Kelly McLaughlin <kelly.mclaughlin@joyent.com>
Date: Fri, 6 Oct 2017 09:25:16 -0600
Subject: [PATCH] MANATEE-323 manatee misguidedly convinced that if it calls
 watch() on a dead ZK client enough, it will come back to life

---
 lib/zookeeperMgr.js | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/lib/zookeeperMgr.js b/lib/zookeeperMgr.js
index 7fcfba5..55ef126 100644
--- a/lib/zookeeperMgr.js
+++ b/lib/zookeeperMgr.js
@@ -247,7 +247,9 @@ function watch(zk, getFunc, path, wat, cb) {
                 path: path,
                 err: err
             }, 'zk: error fetching zk data or children');
-            setTimeout(registerWatch, 5000);
+            if (err.code !== zkClient.Exception.CONNECTION_LOSS) {
+                setTimeout(registerWatch, 5000);
+            }
         } else {
             wat.call(self, watRes);
         }
-- 
2.21.0

